source/h2_connection.c

```diff
@@ -521,13 +529,77 @@ static int s_decoder_on_ping(uint8_t opaque_data[AWS_H2_PING_DATA_SIZE], void *u
     }
 
     aws_h2_connection_enqueue_outgoing_frame(connection, ping_ack_frame);
-    s_try_write_outgoing_frames(connection);
     return AWS_OP_SUCCESS;
 error:
     CONNECTION_LOGF(ERROR, connection, "Ping ACK frame failed to be sent, error %s", aws_error_name(aws_last_error()));
     return AWS_OP_ERR;
 }
 
+static void s_aws_h2_decoder_change_settings(struct aws_h2_connection *connection) {
+    struct aws_h2_decoder *decoder = connection->thread_data.decoder;
+    uint32_t *settings_self = connection->thread_data.settings_self;
+    aws_h2_decoder_set_setting_header_table_size(decoder, settings_self[AWS_H2_SETTINGS_HEADER_TABLE_SIZE]);
+    aws_h2_decoder_set_setting_enable_push(decoder, settings_self[AWS_H2_SETTINGS_ENABLE_PUSH]);
+    aws_h2_decoder_set_setting_max_frame_size(decoder, settings_self[AWS_H2_SETTINGS_MAX_FRAME_SIZE]);
+}
+
+static void s_aws_h2_connection_change_peer_settings(
+    struct aws_h2_connection *connection,
+    struct aws_h2_frame_setting setting) {
+    uint32_t *settings_peer = connection->thread_data.settings_peer;
+    settings_peer[setting.id] = setting.value;
+}
```

Comment:
I don't know, I just think it may have more functions, like informing others? But since the encoder is the only one who will be informed, it is not needed to be a function